packages/block-library/src/site-logo/edit.js

```diff
@@ -349,14 +403,57 @@ export default function LogoEdit( {
 				alt: mediaItem.alt_text,
 			},
 			isRequestingMediaItem: _isRequestingMediaItem,
+			siteIconId: _siteIconId,
+			logoBlockCount: getGlobalBlockCount( 'core/site-logo' ),
```

Comment:
Good catch -- I must have only been testing removing blocks in the order they were inserted! This bug makes sense because I was counting the `logoBlockCount` only on the first render.

I've updated the PR to calculate the blockCount _at the time of removal_.